Tasks/SonarQubePostTest/SonarQubeHelper.ps1

```diff
@@ -0,0 +1,93 @@
+function InvokeMsBuildRunnerPostTest
+{
+	$bootstrapperPath = GetBoostrapperPath
+	$arguments = GetMsBuildRunnerPostTestArgs
+
+	Write-Verbose -Verbose "Executing $bootstrapperPath with arguments $arguments"
+	Invoke-BatchScript $bootstrapperPath -Arguments $arguments
+}
+
+function GetBoostrapperPath
+{
+	$bootstrapperPath = GetTaskContextVariable "bootstrapperPath" 
+
+	if (!$bootstrapperPath -or ![System.IO.File]::Exists($bootstrapperPath))
+	{
+		throw "The SonarQube MsBuild Runner executable could not be found. Does your build definition include the SonarQube Pre-Build step?"
+	}
+
+	Write-Verbose -Verbose "bootstrapperPath: $bootstrapperPath"
+	return $bootstrapperPath;
+}
+
+#
+# Remarks: Normally all the settings are stored in a file on the build agent, but some well-known sensitive settings need to 
+# be passed again as they cannot be stored in non-encrypted files
+#
+function GetMsBuildRunnerPostTestArgs()
+{
+	  $serverUsername = GetTaskContextVariable "serverUsername" 
+	  $serverPassword = GetTaskContextVariable "serverPassword" 
+	  $dbUsername = GetTaskContextVariable "dbUsername" 
+	  $dbPassword = GetTaskContextVariable "dbPassword" 
+
+	  $sb = New-Object -TypeName "System.Text.StringBuilder"; 
+      [void]$sb.Append("end");
+
+	
+      if (![String]::IsNullOrWhiteSpace($serverUsername))
+      {
+          [void]$sb.Append(" /d:sonar.login=""$serverUsername""")
+      }
+	  
+      if (![String]::IsNullOrWhiteSpace($serverPassword))
+      {
+          [void]$sb.Append(" /d:sonar.password=""$serverPassword""")
+      }
+	  
+	   if (![String]::IsNullOrWhiteSpace($dbUsername))
+      {
+          [void]$sb.Append(" /d:sonar.jdbc.username=""$dbUsername""")
+      }
+	  
+      if (![String]::IsNullOrWhiteSpace($dbPassword))
+      {
+          [void]$sb.Append(" /d:sonar.jdbc.password=""$dbPassword""")
+      }
+
+	return $sb.ToString();
+}
+
+function UploadSummaryMdReport
```

Comment:
The online diff-ing experience in GitHub is quite basic for code that has been moved, so if you've moved code and not changed it please call this out in a review comment to assist reviewers.
If you both moved and changed code then call out the changes in comments.